Reordered composer requirements

jordisala1991 opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 1.x is for everything backwards compatible, like patches, features and deprecation notices
    - master is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/cache/blob/1.x/CONTRIBUTING.md#the-base-branch
-->
I am targeting this branch, because this applies for this new release.
Also this could be done for 1.x so maybe we need to move this PR to 1.x branch first.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Fixed
- Add PHP and Psr/log as required dependencies
```
## Subject

<!-- Describe your Pull Request content here -->
At first I wanted to update PHP from 7.0 to 7.1, then I saw the other things:

* Add Sonata contributors as authors
* Move PHP to required section
* Move psr/log to required section (not sure about this one)
* Sort composer packages

greg0ire commented:
You should have done one commit per item in your list, using `git add -p` Also "reored" => "reordered"

greg0ire commented:
`psr/log` was introduced in 7e09191a102143164294454abe1e92c65363a9da, and no reason was given for doing that. I think it is safe to move it to the `require` section without giving it too much thought though, because it's a very common and lightweight dependency to have.

greg0ire commented:
> Remove autoload-dev (was removed on AdminBundle a long ago)

Please explain why you did that in the commit message. I think it is because phpunit does not need autoload to work.

core23 commented on composer.json:
```diff
@@ -32,13 +38,13 @@
     "autoload": {
         "psr-4": { "Sonata\\Cache\\": "src/" }
     },
-    "autoload-dev": {
```
Why did you remove this?

jordisala1991 replied:
Same reason it was removed here: https://github.com/sonata-project/SonataAdminBundle/pull/4295

core23 replied:
You're right

core23 commented on composer.json:
```diff
@@ -10,17 +10,23 @@
         "doctrine/orm": "^2.5",
-        "predis/predis": "^1.1",
         "doctrine/phpcr-odm": "^1.4",
         "jackalope/jackalope-doctrine-dbal": "^1.2",
```
Do we need this? Couldn't find any usage of `Jackalope`

jordisala1991 replied:
https://github.com/jackalope/jackalope-doctrine-dbal

this is in order to use phpcr, as you can see on suggest of phpcr-odm:

https://github.com/doctrine/phpcr-odm/blob/master/composer.json

jordisala1991 commented:
Will re-do my commits in a better way asap

jordisala1991 commented:
Now the commits should be better, please review @sonata-project/contributors

greg0ire commented:
The build fails though

jordisala1991 commented:
You're right, we can remove autoload-dev only if autoload already covers those directories. On this repository we are using src / tests structure which is different from the Sonata Bundles.

Removed that commit, not it should pass.

Also added some phpunit.xml.dist modifications just to see if it changes something on the coverage report

greg0ire commented:
> we can remove autoload-dev only if autoload already covers those directories

No, the right rule is remove it if you don't need to autoload any test class. Here, we extend a tests class `BaseTest` or whatever, and we need to autoload it. If there was no base test class, no autoload would work just fine.

OskarStark commented:
Thank you @jordisala1991 👍
